cmd/influx/task.go

```diff
@@ -3,342 +3,425 @@ package main
+	registerPrintOptions(t.opts.viper, cmd, &t.taskPrintFlags.hideHeaders, &t.taskPrintFlags.json)
+	cmd.Flags().StringVarP(&t.taskID, "id", "i", "", "task ID")
+	cmd.Flags().StringVarP(&t.taskFindFlags.user, "user-id", "n", "", "task owner ID")
+	cmd.Flags().IntVarP(&t.taskFindFlags.limit, "limit", "", influxdb.TaskDefaultPageSize, "the number of tasks to find")
+	cmd.Flags().BoolVar(&t.taskFindFlags.headers, "headers", true, "To print the table headers; defaults true")
 
 	return cmd
 }
 
-func taskFindF(cmd *cobra.Command, args []string) error {
-	if err := taskFindFlags.org.validOrgFlags(&flags); err != nil {
+func (t *cmdTaskBuilder) taskFindF(cmd *cobra.Command, args []string) error {
+
+	if err := t.org.validOrgFlags(&flags); err != nil {
 		return err
 	}
 
-	client, err := newHTTPClient()
+	tskSvc, _, err := t.svcFn()
 	if err != nil {
 		return err
 	}
 
-	s := &http.TaskService{
-		Client: client,
-	}
-
 	filter := influxdb.TaskFilter{}
-	if taskFindFlags.user != "" {
-		id, err := influxdb.IDFromString(taskFindFlags.user)
+	if t.taskFindFlags.user != "" {
+		id, err := influxdb.IDFromString(t.taskFindFlags.user)
 		if err != nil {
 			return err
 		}
 		filter.User = id
 	}
 
-	if taskFindFlags.org.name != "" {
-		filter.Organization = taskFindFlags.org.name
+	if t.org.name != "" {
+		filter.Organization = t.org.name
 	}
-	if taskFindFlags.org.id != "" {
-		id, err := influxdb.IDFromString(taskFindFlags.org.id)
+	if t.org.id != "" {
+		id, err := influxdb.IDFromString(t.org.id)
 		if err != nil {
 			return err
 		}
 		filter.OrganizationID = id
 	}
 
-	if taskFindFlags.limit < 1 || taskFindFlags.limit > influxdb.TaskMaxPageSize {
+	if t.taskFindFlags.limit < 1 || t.taskFindFlags.limit > influxdb.TaskMaxPageSize {
 		return fmt.Errorf("limit must be between 1 and %d", influxdb.TaskMaxPageSize)
 	}
-	filter.Limit = taskFindFlags.limit
+	filter.Limit = t.taskFindFlags.limit
 
 	var tasks []*influxdb.Task
 
-	if taskFindFlags.id != "" {
-		id, err := influxdb.IDFromString(taskFindFlags.id)
+	if t.taskID != "" {
+		id, err := influxdb.IDFromString(t.taskID)
 		if err != nil {
 			return err
 		}
 
-		task, err := s.FindTaskByID(context.Background(), *id)
+		task, err := tskSvc.FindTaskByID(context.Background(), *id)
 		if err != nil {
 			return err
 		}
 
 		tasks = append(tasks, task)
 	} else {
-		tasks, _, err = s.FindTasks(context.Background(), filter)
+		tasks, _, err = tskSvc.FindTasks(context.Background(), filter)
 		if err != nil {
 			return err
 		}
 	}
 
-	return printTasks(
-		cmd.OutOrStdout(),
-		taskPrintOpts{
-			hideHeaders: taskPrintFlags.hideHeaders,
-			json:        taskPrintFlags.json,
-			tasks:       tasks,
-		},
-	)
+	return t.printTasks(taskPrintOpts{tasks: tasks})
 }
 
-var taskUpdateFlags struct {
-	id     string
+type taskRerunFailedFlags struct {
+	before string
+	after  string
+}
+
+func (t *cmdTaskBuilder) taskRerunFailedCmd() *cobra.Command {
+	cmd := t.opts.newCmd("rerun_failed", t.taskRerunFailedF, true)
```

Comment:
More conventional for unix commands, and matches our other multi-word subcommands:
```suggestion
	cmd := t.opts.newCmd("rerun-failed", t.taskRerunFailedF, true)
```

Reply:
Alt: would `retry` capture the same meaning as `rerun-failed`?

Reply:
Ah, looks like it's already handled.